SECP-service/src/main/java/com/visucius/secp/Controllers/GroupController.java

```
@@ -0,0 +1,136 @@
+import com.visucius.secp.util.JsonUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class GroupController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GroupController.class);
+    public static final int MINIMUM_AMOUNT_OF_USERS = 2;
+    public static final int MAXIMUM_AMOUNT_OF_USERS = 50;
+
+    private GroupDAO groupRepository;
+    private UserDAO userRepository;
+    private RolesDAO rolesRepository;
+
+    public GroupController(GroupDAO groupRepository, UserDAO userRepository, RolesDAO rolesRepository)
+    {
+        this.groupRepository = groupRepository;
+        this.userRepository = userRepository;
+        this.rolesRepository = rolesRepository;
+    }
+
+
+    public Response createGroup(GroupCreationRequest request)
+    {
+        List<String> errors = validateCreationRequest(request);
+
+        if(!errors.isEmpty())
+        {
+            try
+            {
+                throw new WebApplicationException(
+                    JsonUtil.convertToJsonString(errors),
+                    Response.Status.BAD_REQUEST);
+            }
+            catch (JsonProcessingException exception)
+            {
+                LOG.error(exception.getMessage());
+                throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
+            }
+        }
+
+        Group group = new Group(request.name);
+        group.setRoles(getRoles(request.roles));
+        group.setUsers(getUsers(request.users));
+        Group createdGroup = groupRepository.save(group);
+
+        return Response.status(Response.Status.CREATED).entity(createdGroup.getId()).build();
+    }
+
+
+    private List<String> validateCreationRequest(GroupCreationRequest request)
+    {
+        List<String> errors = new ArrayList<>();
```

Please throw a single error here. We display the error directly from the server on the client with a swal. Having multiple errors won't look good.